src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.swr

```diff
@@ -7,10 +7,10 @@ vs.dependencies
   vs.dependency id=Microsoft.Net.PackageGroup.4.6.1.Redist
 
 folder InstallDir:\MSBuild\15.0\Bin\Roslyn
-    file source=$(OutputPath)\Exes\VBCSCompiler\VBCSCompiler.exe vs.file.ngenArchitecture=all
-    file source=$(OutputPath)\Exes\csc\csc.exe vs.file.ngenArchitecture=all
-    file source=$(OutputPath)\Exes\csi\csi.exe vs.file.ngenArchitecture=all
-    file source=$(OutputPath)\Exes\vbc\vbc.exe vs.file.ngenArchitecture=all
+    file source=$(OutputPath)\Exes\VBCSCompiler\VBCSCompiler.exe vs.file.ngenArchitecture=all vs.file.ngenPriority=1
+    file source=$(OutputPath)\Exes\csc\csc.exe vs.file.ngenArchitecture=all vs.file.ngenPriority=1
+    file source=$(OutputPath)\Exes\csi\csi.exe vs.file.ngenArchitecture=all vs.file.ngenPriority=1
+    file source=$(OutputPath)\Exes\vbc\vbc.exe vs.file.ngenArchitecture=all vs.file.ngenPriority=1
```

Comment:
sigh ... why is this not the default

Reply:
I've thought the same thing myself, but I have to concede that it's a slippery slope...

At some point, your install experience will suffer.  It's a better experience overall if ngen of *most* dlls loaded in VS is delayed, but it's tough to get people to voluntarily downgrade the priority of their assemblies, hence the tactic performance through obscurity.

Reply:
❓ Are all of these executables directly affected? Or is one more so than others?

It seems like **csi.exe** could be priority 2, and it's possible that **VBCSCompiler.exe** is actually the only one that needs to be priority 1.

Reply:
The problem (mostly) is that they all reference a bagillion façade assemblies (via Microsoft.CodeAnalysis.dll).  Once the binary is ngen'ed, we don't even need to load the facades to resolve the type forwarders, but until that happens, we have to load and resolve the type forwarders again and again while jit'ing.

One *could* argue that csi is somewhat less important, but I personally don't think it's worth it to save a few seconds of setup cost (because we somewhat frequently get anecdotal complaints that csi is too slow).  In the past, it was pri 1.